### PR TITLE
[SOAR-10497] - TypeConverter error handling

### PR DESCRIPTION
**Pull request by cmcnally-r7**
### Description

Describe the proposed changes:

  - Fixed error handling for string_to_int & string_to_float such that the errors are displayed client-side.
  - Update 'e' variable to 'error' across all actions for PluginException.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [x] Screenshot of job output with the plugin changes
- [x] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder
Screenshot 2022-08-02 at 10 38 26

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [x] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [x] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [x] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [x] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [x] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [x] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [x] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [x] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [x] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Review by TKuzas-R7 — APPROVED** [review]

File: plugins/type_converter/help.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ for enabling input interoperability between certain plugins.
 
 _This plugin does not contain any requirements._
 
+# Supported Product Versions
+
+_There are no supported product versions listed._
+
```
Comment (ablakley-r7) [inline plugins/type_converter/help.md:17]: Just a general question, but wondering if this is required?

**Review by ablakley-r7 — COMMENTED** [review]
I think the PluginExceptions in all other actions could use the same update. For instance, having a similar problem with String to Object action.

File: plugins/type_converter/help.md (hunk 1 above)
Comment (cmcnally-r7) [reply plugins/type_converter/help.md:17]: It got added when I regenerated. Is it good practice to remove this?

Comment (cmcnally-r7) [reply plugins/type_converter/help.md:17]: I'll have a look through and replicate for each of the other actions too. Good spot.

**Review by jkaluza-rapid7 — APPROVED** [review]

**Review by igorski-r7 — APPROVED** [review]

**Review by jkaluza-rapid7 — COMMENTED** [review]
Yes, it is not backwards compatible for sure, we should discuss with others how do they want it to work.

**Review by llaszuk-r7 — APPROVED** [review]

**Pull request merged**